File: TypeScript/problem005.ts
/*
 * Problem 5 - Project Euler
 * http://projecteuler.net/index.php?section=problems&id=5
 */
{
    const gcd = (x: number, y: number): number => {
        return x === 0 ? y : gcd(y % x, x);
    };

    const lcm = (x: number, y: number): number => {
        return (x * y) / gcd(x, y);
    };

    console.log(
        [...Array(21).keys()].slice(1).reduce(lcm, 1)
    );
}

File: Kotlin/src/main/kotlin/problem005.kt
/*
 * Problem 5 - Project Euler
 * http://projecteuler.net/index.php?section=problems&id=5
 */
fun gcd(x: Long, y: Long): Long {
  return if (x == 0L) {
    y
  } else {
    gcd(y % x, x)
  }
}

fun lcm(x: Long, y: Long): Long {
  return x * y / gcd(x, y)
}

fun main() {
  println((1L..20).fold(1L, { x, y -> lcm(x, y) }))
}
